.github/workflows/stackhpc-container-image-build.yml

```diff
@@ -200,18 +207,58 @@ jobs:
       - name: Fail if no images have been built
         run: if [ $(wc -l < ${{ matrix.distro }}-container-images) -le 1 ]; then exit 1; fi
 
-      - name: Upload container images artifact
+      - name: Scan built container images
+        run: src/kayobe-config/tools/scan-images.sh ${{ matrix.distro }} ${{ needs.generate-tag.outputs.kolla_tag }}
+
+      - name: Upload Trivy scan results artifact
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ matrix.distro }} container images
-          path: ${{ matrix.distro }}-container-images
+          name: ${{ matrix.distro }}-image-scan-output
+          path: image-scan-output
           retention-days: 7
 
+      - name: Fail if no images have passed scanning
+        run: if [ $(wc -l < image-scan-output/clean-images.txt) -le 0 ]; then exit 1; fi
+        if: github.event.inputs.scan-push == 'false'
+
+      - name: Append dirty images to clean list
+        run: |
+          cat image-scan-output/dirty-images.txt >> image-scan-output/clean-images.txt
+        if: github.event.inputs.scan-push == 'true'
+
+      - name: Push images
+        run: |
+          source venvs/kayobe/bin/activate &&
+          source src/kayobe-config/kayobe-env --environment ci-builder &&
+          kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/docker-registry-login.yml &&
+
+          while read -r image; do
+            # Retries!
+            for i in {1..10}; do 
+              docker push ${image} && break || sleep 5
+            done
+          done < image-scan-output/clean-images.txt
+        shell: bash
+        env:
+          KAYOBE_VAULT_PASSWORD: ${{ secrets.KAYOBE_VAULT_PASSWORD }}
+        if: github.event.inputs.push == 'true'
+
+      - name: Upload pushed container images artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.distro }}-pushed-container-images
+          path: image-scan-output/clean-images.txt
+          retention-days: 7
+
+      - name: Fail when images failed to build
+        run: exit 1
+        if: steps.build_overcloud_images.outcome == 'failure' || steps.build_seed_images.outcome == 'failure'
+
   sync-container-repositories:
     name: Trigger container image repository sync
     needs:
       - container-image-build
-    if: github.repository == 'stackhpc/stackhpc-kayobe-config' && inputs.push
+    if: github.repository == 'stackhpc/stackhpc-kayobe-config' && inputs.push == 'true'
```

Comment:
Are you sure it's necessary? IIRC if you use `inputs.push` you can treat it as a boolean, while if you use `github.event.inputs.push` it's a string...

Reply:
You know I'd always wondered why it sometimes worked and sometimes didn't